presentation/common/src/presentation-common/content/PropertyValueFormatter.ts

```diff
@@ -168,28 +173,31 @@ export class ContentPropertyValueFormatter {
       return value.label.displayValue;
     }
 
+    if (type.typeName === "enum" && field.isPropertiesField()) {
+      return field.properties[0].property.enumerationInfo?.choices.find(({ value: enumValue }) => enumValue === value)?.label;
+    }
     // eslint-disable-next-line @typescript-eslint/no-base-to-string
     return value.toString();
   }
 
-  private async formatStructValue(type: StructTypeDescription, value: Value) {
+  private async formatStructValue(type: StructTypeDescription, value: Value, field: Field) {
     if (!Value.isMap(value)) {
       return {};
     }
 
     const formattedMember: DisplayValuesMap = {};
     for (const member of type.members) {
-      formattedMember[member.name] = await this.formatValue(member.type, value[member.name]);
+      formattedMember[member.name] = await this.formatValue(member.type, value[member.name], field);
```

Comment:
Here you're formatting a struct member, but passing struct field - that's not going to work. I think you should ensure `field` is `StructPropertiesField` and find appropriate member field to pass to `formatValue`.

Same problem with array items below.  Please add tests for struct members and array items.

Reply:
Changed the workflow for struct and array formatting, still need to adjust a few test.

Reply:
Fixed